fluent/client/client_test.go

```diff
@@ -53,11 +53,17 @@ var _ = Describe("Client", func() {
 			Expect(factory.NewCallCount()).To(Equal(1))
 		})
 
-		It("Stores the connection in the Session", func() {
+		It("Completes the handshake", func() {
+			err := client.Connect()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(client.TransportPhase()).To(BeTrue())
+		})
+
+		It("errors if session already exists", func() {
 			err := client.Connect()
 			Expect(err).NotTo(HaveOccurred())
-			Expect(client.Session).NotTo(BeNil())
-			Expect(client.Session.Connection).To(Equal(clientSide))
+			err = client.Connect()
+			Expect(err.Error()).To(Equal("a session is already active"))
```

Comment:
Just a suggestion - in case of comparing error strings it might be an option to provide a dedicated error element/constant instance witch enables use of standard go library function `errors.Is()` or `errors.As()`.